File: .agent/skills/cnma-cap-backend/templates/java/srv/src/main/java/cnma/{{module_name}}/domain/__feature__/events/OnBeforeUpdate{{FeatureName}}Event.java
package cnma.{{module_name}}.domain.__feature__.events;

import com.sap.cds.services.RequestContext;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

/**
 * OnBeforeUpdate{{FeatureName}}Event - BEFORE UPDATE event handler.
 * SRP: Validates data before update, sets audit fields.
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class OnBeforeUpdate{{FeatureName}}Event {

    private final RequestContext requestContext;

    public void execute(Object req) {
        String userId = getUserId(req);
        log.debug("BEFORE UPDATE {{FeatureName}} by user: {}", userId);

        // Set audit fields (modifiedBy, modifiedAt)
        // Validate data
        // Check if record exists
    }

    private String getUserId(Object req) {
        return "anonymous";
    }
}

File: .agent/skills/cnma-cap-backend/templates/node/srv/src/domain/__feature__/events/OnBeforeUpdate{{FeatureName}}Event.js
import cds from '@sap/cds';

/**
 * OnBeforeUpdate{{FeatureName}}Event - BEFORE UPDATE event handler.
 * SRP: Validates data before update.
 */
module.exports = {
    execute: async (req) => {
        try {
            const userId = req.user?.id || 'anonymous';
            const data = req.data;

            // Set audit field
            data.modifiedBy = userId;
            data.modifiedAt = new Date();

            // Validation logic here
            if (!data || Object.keys(data).length === 0) {
                req.error(400, 'Invalid data: empty payload');
            }

            // Check if record exists
            const recordId = req.params?.[0];
            if (recordId) {
                const existing = await cds.run(
                    cds.select('{{FeatureName}}').where({ ID: recordId })
                );
                if (!existing || existing.length === 0) {
                    req.error(404, 'Record not found');
                }
            }
        } catch (error) {
            req.error(500, 'Error in before update: ' + error.message);
        }
    }
};